prov/shm/src/smr_progress.c

```diff
@@ -319,9 +321,26 @@ static int smr_progress_iov(struct smr_cmd *cmd, struct iovec *iov,
 		goto out;
 	}
 
-	ret = smr_cma_loop(peer_smr->pid, iov, iov_count, cmd->msg.data.iov,
-			   cmd->msg.data.iov_count, 0, cmd->msg.hdr.size,
-			   cmd->msg.hdr.op == ofi_op_read_req);
+	/* TODO: Once cma is migrated to the new shm p2p infrastructure,
+	 * we should be able to call ofi_shm_p2p_copy() with the
+	 * appropriate p2p_type, instead of having a separate function
+	 * call for cma
+	 */
+	xpmem = &smr_peer_data(ep->region)[cmd->msg.hdr.id].xpmem;
+	if (ep->region->xpmem_cap_self == SMR_VMA_CAP_ON &&
+	    xpmem->cap == SMR_VMA_CAP_ON) {
```

Comment:
why are we checking this on the receive/progress side? shouldn't the sender be checking these and only sending if xpmem is available on the peer?

Reply:
We'd need to do that check anyway to make the proper call to copy using XPMEM or otherwise. Am I missing something?
Basically the address space of a process is exposed via XPMEM. A pointer to the address is passed over to the receiving process. the receiving process figures that it and the sender are using XPMEM and then makes the appropriate call into p2p infra.

Reply:
just to clarify, on the sending side, there is no need to check XPMEM, because it works exactly as CMA. The local buffer address is passed to the receiver. The Receiver then checks if XPMEM is enabled locally and on the sender. If so, then it knows that the address space of the remote process is now exported via XPMEM. It can then safely attach to that memory and copy into the local buffers.

Reply:
Gotcha. Thanks for clarifying. I thought there was something extra that needed to happen on the sender side